### PR TITLE
Fixes #18317: Support source package in Rudder inventory

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/18317

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

File: webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/SoftwareIdFinder.scala [hunk 1]
```diff
@@ -90,7 +90,7 @@ class NameAndVersionIdFinder(
     } yield {
       //now merge back
       entities.foldLeft(MergedSoftware(Set(),Set())) { case(ms, s) =>
-        merged.find { x => x.name == s.name && x.version == s.version } match {
+        merged.find { x => x == s } match {
```
Comment (fanf) [inline webapp/sources/ldap-inventory/inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/SoftwareIdFinder.scala:93]: I don't think we want to compare ALL fields ? Only name, version, source name, source version ?

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

File: webapp/sources/ldap-inventory/inventory-repository/src/main/resources/ldap/inventory.schema [hunk 2]
```diff
@@ -1078,7 +1078,7 @@ objectclass ( InventoryObjectClasses:21
   STRUCTURAL
   MUST ( softwareId )
   MAY ( cn $ description $ releaseDate $ editor $ softwareVersion $
-        licenseExpirationDate $ licenseName $ licenseProductId $ licenseProductKey ) )
+        licenseExpirationDate $ licenseName $ licenseProductId $ licenseProductKey $ sourceName $ sourceVersion ) )
```
Comment (fanf) [inline webapp/sources/ldap-inventory/inventory-repository/src/main/resources/ldap/inventory.schema:1081]: `sourceName` and `sourceVersion` attributeType must be defined before use

**Review by fanf — CHANGES_REQUESTED** [review]

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by VinceMacBuche** [comment]
OK, squash merging this PR

**Pull request merged**